pkg/ext/execution_fetcher.go

```diff
@@ -23,14 +22,32 @@ func (a *AdminFetcherExtClient) FetchExecution(ctx context.Context, name, projec
 	return e, nil
 }
 
-func (a *AdminFetcherExtClient) FetchNodeExecutionDetails(ctx context.Context, name, project, domain string) (*admin.NodeExecutionList, error) {
+func (a *AdminFetcherExtClient) FetchNodeExecutionData(ctx context.Context, nodeID, execName, project, domain string) (*admin.NodeExecutionGetDataResponse, error) {
+	ne, err := a.AdminServiceClient().GetNodeExecutionData(ctx, &admin.NodeExecutionGetDataRequest{
+		Id: &core.NodeExecutionIdentifier{
+			NodeId: nodeID,
+			ExecutionId: &core.WorkflowExecutionIdentifier{
+				Project: project,
+				Domain:  domain,
+				Name:    execName,
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return ne, nil
+}
+
+func (a *AdminFetcherExtClient) FetchNodeExecutionDetails(ctx context.Context, name, project, domain, uniqueParentId string) (*admin.NodeExecutionList, error) {
 	ne, err := a.AdminServiceClient().ListNodeExecutions(ctx, &admin.NodeExecutionListRequest{
 		WorkflowExecutionId: &core.WorkflowExecutionIdentifier{
 			Project: project,
 			Domain:  domain,
 			Name:    name,
 		},
-		Limit: 100,
+		UniqueParentId: uniqueParentId,
+		Limit:          100,
```

Comment:
what if number of nodes is more than 100?

Reply:
We are going to support pagination across all flyte entities where we will address this 
https://github.com/flyteorg/flyte/issues/1146